src/General/Badge/Badge.test.tsx

```diff
@@ -0,0 +1,36 @@
+import * as React from 'react';
+
+import * as renderer from 'react-test-renderer';
+import '@testing-library/jest-dom/extend-expect';
+import { render } from '@testing-library/react';
+
+import Badge from './Badge';
+
+const label = '6';
+
+it(`<Badge> should render a label with the text "6"`, () => {
+  const BadgeSnapshot = renderer.create(<Badge label={label} />).toJSON();
+  expect(BadgeSnapshot).toMatchSnapshot();
+});
+
+it('should display the correct text when rendered', () => {
+  const { getByText } = render(<Badge label={label} />);
+  const BadgeContent = getByText(label);
+  expect(BadgeContent).toBeTruthy();
```

Comment:
`expect(BadgeContent).toBeInTheDocument()` or `expect(BadgeContent).toBeVisible()`  is more declarative, and `getByText` will throw error if no element is found, so I use `queryByXXX` more, which returns `null` when no element found.